From 62ae17984e25819cc0aae942efb203c62f6bee91 Mon Sep 17 00:00:00 2001
From: Anson Huang <Anson.Huang@nxp.com>
Date: Wed, 8 Mar 2017 00:48:42 +0800
Subject: [PATCH 1553/5242] MLK-14359-2 soc: imx8: skip power-domain
 initialization for non-SCFW system

commit  1ceb57d3abbca61355e16651ffd86b33d36cba69 from
https://source.codeaurora.org/external/imx/linux-imx.git

For non-SCFW system, the power domain initialization is NOT
necessary, so add check to skip it if necessary.

Signed-off-by: Anson Huang <Anson.Huang@nxp.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 drivers/soc/imx/pm-domains.c |    4 ++++
 1 file changed, 4 insertions(+)

diff --git a/drivers/soc/imx/pm-domains.c b/drivers/soc/imx/pm-domains.c
index 51202b7..158ceb6 100644
--- a/drivers/soc/imx/pm-domains.c
+++ b/drivers/soc/imx/pm-domains.c
@@ -100,6 +100,10 @@ static int __init imx8_init_pm_domains(void)
 	sc_rsrc_t rsrc_id;
 	uint32_t mu_id;
 
+	/* skip pm domains for non-SCFW system */
+	if (!of_find_compatible_node(NULL, NULL, "nxp,imx8-pd"))
+		return 0;
+
 	pr_info("***** imx8_init_pm_domains *****\n");
 
 	for_each_compatible_node(np, NULL, "nxp,imx8-pd") {
-- 
1.7.9.5

